docs/src/LeNetTraining.md

```diff
@@ -0,0 +1,123 @@
+# Training a LeNet architecture with JSO optimizers
+
+## Neural network architecture
+Define the layer Julia structures, how these structures are linked and evaluated.
+We choose the negative log likelihood loss function.
+
+```@example LeNetTraining
+using Knet
+
+struct ConvolutionnalLayer
+  weight
+  bias
+  activation_function
+end
+# evaluation of a ConvolutionnalLayer layer given an input x
+(c::ConvolutionnalLayer)(x) = c.activation_function.(pool(conv4(c.weight, x) .+ c.bias))
+# Constructor of a ConvolutionnalLayer structure
+ConvolutionnalLayer(kernel_width, kernel_height, channel_input, 
+  channel_output, activation_function = relu) = 
+  ConvolutionnalLayer(
+    param(kernel_width, kernel_height, channel_input, channel_output), 
+    param0(1, 1, channel_output, 1), 
+    activation_function
+  )
+
+struct DenseLayer
+  weight
+  bias
+  activation_function
+end
+# evaluation of a DenseLayer given an input x
+(d::DenseLayer)(x) = d.activation_function.(d.weight * mat(x) .+ d.bias)
+# Constructor of a DenseLayer structure
+DenseLayer(input::Int, output::Int, activation_function = sigm) =
+  DenseLayer(param(output, input), param0(output), activation_function)
+
+# A chain of layers ended by a negative log likelihood loss function
+struct Chainnll
+  layers
+  Chainnll(layers...) = new(layers) # Chainnll constructor
+end
+# Evaluate successively each layer
+# A layer's input is the precedent layer's output
+(c::Chainnll)(x) = (for l in c.layers
+  x = l(x)
+end;
+x)
+# Apply the negative log likelihood function 
+# on the result of the neural network forward pass
+(c::Chainnll)(x, y) = Knet.nll(c(x), y)
+(c::Chainnll)(data::Tuple{T1, T2}) where {T1, T2} = c(first(data, 2)...)
+(c::Chainnll)(d::Knet.Data) = Knet.nll(c; data = d, average = true)
+
+output_classes = 10
+
+LeNet = Chainnll(ConvolutionnalLayer(5,5,1,6), 
+                 ConvolutionnalLayer(5,5,6,16),
+                 DenseLayer(256, 120),
+                 DenseLayer(120, 84),
+                 DenseLayer(84,output_classes)
+                )
+```
+
+## MNIST dataset loading
+Accordingly to LeNet architecture, we chose the MNIST dataset [@lecun-bouttou-bengio-haffner1998] from MLDataset:
```

Comment:
Does it find the reference or we need to add references manually ?

Reply:
It will not find it.
Thanks for the catch, I will fix this soon.